modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -1180,6 +1199,88 @@
     (for [i (range (apply max (map count posts)))]
       (into {} (map #(get % i)) posts))))
 
+(mu/defn- order-by->$sort :- [:map-of ::lib.schema.common/non-blank-string [:enum -1 1]]
+  [order-by :- [:sequential ::mbql.s/OrderBy]]
+  (into
+   (ordered-map/ordered-map)
+   (for [[direction field] order-by]
+     [(->lvalue field) (case direction
+                         :asc   1
+                         :desc -1)])))
+
+(defn- generate-window-output-clause
+  "Takes a pair of [output-name input-name] and generates an output clause suitable for
+  including in a `$setWindowFields` output block."
+  [[output-name input-name]]
+  {output-name
+   {$sum input-name
+    "window" {"documents" ["unbounded" "current"]}}})
+
+(defn- sort-lookup
+  "Generates a lookup string for a particular field"
+  [id name]
+  (if (id name)
+    (str "_id." name)
+    name))
+
+(defn- make-window-sort
```

Comment:
A naming thing, for this and some other functions here, we don't generally have verbs at the front unless there are side effects. For pure transformations or synthesizing data structures, prefer nouns: `window-sort`, `window-sort-and-partitions`, `window-output-clause`, `window-accumulators`.

Reply:
sure